database/mysql_user.py

```diff
@@ -155,6 +160,30 @@
 # MySQL module specific support methods.
 #
 
+def connect(module, login_user, login_password, config_file):
+    default_file = '~/.my.cnf'
+    if config_file is not None:
+        default_file = config_file
```

Comment:
This might be much simpler if you just use:
def connect(module, login_user, login_password, config_file='~/.my.cnf'):
and then just use config_file